### PR TITLE
add config option to wittyer module

### PR DESCRIPTION
**Pull request by famosab**
## PR checklist

Closes #XXX 

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [x] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [ ] Follow the naming conventions.
- [ ] Follow the parameters requirements.
- [ ] Follow the input/output options guidelines.
- [ ] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - For modules:
    - [x] `nf-core modules test <MODULE> --profile docker`
    - [ ] `nf-core modules test <MODULE> --profile singularity`
    - [ ] `nf-core modules test <MODULE> --profile conda`
  - For subworkflows:
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile docker`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile singularity`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile conda`

### CODE REVIEW

**Comment by famosab** [comment]
This will only work after this PR is merged: https://github.com/nf-core/test-datasets/pull/1275

**Comment by kubranarci** [comment]
I think there is a misunderstanding on the config file here. Config file should be in json format specifying parameters. It overwrites given parameters in cmd. It shouldn't be used instead of bed file which is a bed file with region stratifications. They wont replace each other.

**Comment by famosab** [comment]
@kubranarci This is what the documentation says:

> Pass in a configuration file via the '-c,--configFile=' command line argument. - The '-c,--configFile=' argument cannot be used in combination with arguments for --bpDistance, --percentDistance, --binSizes, --includedFilters, --excludedFilters, --variantTypes, or --includeBed.

I understand that using the config file replaces the `-b` (`--includeBed`)  flag. I am also a bit confused by that, but this is denoted multiple times in the docs. But I did not succeed in writing a correct check for that so one could try and hand both to the module.

**Comment by kubranarci** [comment]
> @kubranarci This is what the documentation says:
> 
> > Pass in a configuration file via the '-c,--configFile=' command line argument. - The '-c,--configFile=' argument cannot be used in combination with arguments for --bpDistance, --percentDistance, --binSizes, --includedFilters, --excludedFilters, --variantTypes, or --includeBed.
> 
> I understand that using the config file replaces the `-b` (`--includeBed`) flag. I am also a bit confused by that, but this is denoted multiple times in the docs. But I did not succeed in writing a correct check for that so one could try and hand both to the module.

This is pretty odd yea. I also couldn't find enough run examples. Then, it will be probably me to find out if their statement is true or not :D

**Comment by kubranarci** [comment]
I think you should re create the snaps to get rid of errors?

**Comment by famosab** [comment]
Yes or check for different assertions. I'll re-request your review when I am done.

**Review by kubranarci — APPROVED** [review]
LGTM

File: modules/nf-core/wittyer/meta.yml [hunk 1]
```diff
@@ -22,16 +22,20 @@ input:
...
-
   - bed:
       type: file
       description: A BED file specifying regions to be included in the analysis (optional)
       pattern: "*.bed"
+  - config:
+      type: file
+      description: |
+        Config file in json format used to specify per variant type settings.
+        Used in place of include bed arguments. (optional)
```
[the remark this answers is not in the document]
Comment (famosab) [reply modules/nf-core/wittyer/meta.yml:37]: I just copied that from the official docs

File: modules/nf-core/wittyer/main.nf [hunk 2]
```diff
@@ -24,9 +24,18 @@ process WITTYER {
     script:
     def args  = task.ext.args ?: ''
     def prefix = task.ext.prefix ?: "${meta.id}"
+    // if ((bed) && (wittyer_config)) {
+    //     error "Cannot provide both bed file and config file. Please choose one."
```
[the remark this answers is not in the document]
Comment (famosab) [reply modules/nf-core/wittyer/main.nf]: I just copied that from the official docs. But the checks did not work as I wanted.

File: modules/nf-core/wittyer/main.nf [hunk 3]
```diff
@@ -10,7 +10,7 @@ process WITTYER {
     }
 
     input:
-    tuple val(meta), path(query_vcf), path(truth_vcf), path(bed)
+    tuple val(meta), path(query_vcf), path(truth_vcf), path(bed), path(wittyer_config)
```
[the remark this answers is not in the document]
Comment (famosab) [reply modules/nf-core/wittyer/main.nf:13]: with a meta as well? Should we move the bed file to a special input aswell?

File: modules/nf-core/wittyer/main.nf (hunk 2 above)
[the remark this answers is not in the document]
Comment (famosab) [reply modules/nf-core/wittyer/main.nf]: And I removed the commented out stuff afterwards.

**Pull request merged**